test/e2e-sno-disruptive/sno_disruptive_test.go

```diff
@@ -1,7 +1,261 @@
+		}
+	}
+
+	t.Logf("The fallback has been reported on node %v, failed revision is %v", nodeName, failedRevision)
+	return nodeName, failedRevision
+}
+
+func assertKasPodAnnotatedOnNode(t testing.TB, cs clientSet, expectedFailedRevision int32, expectedFallbackReason, expectedFallbackMessage, nodeName string) {
+	t.Helper()
+	t.Logf("Verifying if a kube-apiserver pod has been annotated with rev: %v, reason: %v, message: %v on node: %v", expectedFailedRevision, expectedFallbackReason, expectedFallbackMessage, nodeName)
+
+	kasPods, err := cs.Kube.CoreV1().Pods("openshift-kube-apiserver").List(context.TODO(), metav1.ListOptions{LabelSelector: "apiserver=true"})
+	require.NoError(t, err)
+
+	var kasPod corev1.Pod
+	filteredKasPods := filterByNodeName(kasPods.Items, nodeName)
+	switch len(filteredKasPods) {
+	case 0:
+		t.Fatalf("expected to find the kube-apiserver static pod on node %s but haven't found any", nodeName)
+	case 1:
+		kasPod = filteredKasPods[0]
+	default:
+		// this should never happen for static pod as they are uniquely named for each node
+		podsOnCurrentNode := []string{}
+		for _, filteredKasPod := range filteredKasPods {
+			podsOnCurrentNode = append(podsOnCurrentNode, filteredKasPod.Name)
+		}
+		t.Fatalf("multiple kube-apiserver static pods for node %s found: %v", nodeName, podsOnCurrentNode)
+	}
+
+	if fallbackFor, ok := kasPod.Annotations[annotations.FallbackForRevision]; ok {
+		if len(fallbackFor) == 0 {
+			t.Fatalf("empty fallback revision label: %v on %s pod", annotations.FallbackForRevision, kasPod.Name)
+		}
+		revision, err := strconv.Atoi(fallbackFor)
+		if err != nil || revision < 0 {
+			t.Fatalf("invalid fallback revision: %v on pod: %s", fallbackFor, kasPod.Name)
+		}
+
+		reason, message := "", ""
+		if s, ok := kasPod.Annotations[annotations.FallbackReason]; ok {
+			reason = s
+		}
+		if s, ok := kasPod.Annotations[annotations.FallbackMessage]; ok {
+			message = s
+		}
+
+		if expectedFallbackReason != "*" && expectedFallbackReason != reason {
+			t.Fatalf("unexpected fallback reason: %v, expected: %v", reason, expectedFallbackReason)
+		} else if len(reason) == 0 {
+			t.Fatal("empty fallback reason")
+		}
+		if expectedFallbackMessage != "*" && expectedFallbackMessage != message {
+			t.Fatalf("unexpected fallback message: %v, expected: %v", message, expectedFallbackMessage)
+		} else if len(message) == 0 {
+			t.Fatal("empty fallback message")
+		}
+		return
+	}
+
+	t.Fatalf("kube-apiserver %v on node %v hasn't been annotated with %v", kasPod.Name, nodeName, annotations.FallbackForRevision)
+}
+
+func filterByNodeName(kasPods []corev1.Pod, currentNodeName string) []corev1.Pod {
+	filteredKasPods := []corev1.Pod{}
+
+	for _, potentialKasPods := range kasPods {
+		if potentialKasPods.Spec.NodeName == currentNodeName {
+			filteredKasPods = append(filteredKasPods, potentialKasPods)
+		}
+	}
+
+	return filteredKasPods
+}
+
+func getDefaultUnsupportedConfig(t testing.TB, cs clientSet) map[string][]string {
```

Comment:
not sure about this func. Name is unspecific, and it does abstract something that's better explicit.

Reply:
`getPlatformSpecificConfig` would be better?

Reply:
I think the infra check is nothing one should hide in a func